### PR TITLE
update (3.3.5-solocraft): Maintence

### PR DESCRIPTION
**Pull request by acidmanifesto**
**Changes proposed**:

- maintence, possible fix segment error someone is reporting that i can not seem to replicate.

**Target branch(es)**: 3.3.5-solocraft

**Issues addressed**: Fixes # https://github.com/TrinityCore/TrinityCoreCustomChanges/issues/140

**Tests performed**: (Does it build, tested in-game, etc)

**Known issues and TODO list**:

- [ ] 
- [ ]

### CODE REVIEW

File: src/server/scripts/Custom/Solocraft.cpp [hunk 1]
```diff
@@ -525,7 +525,7 @@ class solocraft_player_instance_handler : public PlayerScript {
             for (Group::member_citerator itr = groupMembers.begin(); itr != groupMembers.end(); ++itr)
             {
                 //Exclude player from the tally because the player is the one entering the dungeon
-                if (itr->guid != player->GetGUID())
+                if (itr->guid != player->GetGUID().GetCounter())
```
Comment (jackpoz) [inline src/server/scripts/Custom/Solocraft.cpp]: This shouldn't be changed as it's not a SQL query

**Review by jackpoz — CHANGES_REQUESTED** [review]

File: src/server/scripts/Custom/Solocraft.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/scripts/Custom/Solocraft.cpp]: Thanks. I corrected.

**Review by kruchor — APPROVED** [review]
The login issue is resolved with this change.Getting in instance (in a group) lead to crash. Being solo is OK. I think there might be an extra modification to make, line 528 : 

```
    526             {
    527                 //Exclude player from the tally because the player is the one entering the dungeon
    528                 if (itr->guid != player->GetGUID)
    529                 {
    530                     //Database query to find difficulty for each group member that is currently in an instance
    531                     QueryResult result = CharacterDatabase.PQuery("SELECT `GUID`, `Difficulty`, `GroupSize` FROM `custom_solocraft_character_stats` WHERE GUID = {}", itr->guid);
```

**Comment by jackpoz** [comment]
Post the crashlog pls

**Comment by kruchor** [comment]
I can't reproduce this issue anymore.

I can enter instance with both players and bots (npcbots) without disconnection/crash. I'll try to perform some more tests.

Patch used : https://patch-diff.githubusercontent.com/raw/TrinityCore/TrinityCoreCustomChanges/pull/141.patch

```
trinity@wotlkVM-dev:~/TC_3.3.5-solocraft$ patch  -p1 < 141.patch
patching file src/server/scripts/Custom/Solocraft.cpp
patching file src/server/scripts/Custom/Solocraft.cpp
patching file src/server/scripts/Custom/Solocraft.cpp
```

```
TC> server debug
TC> TrinityCore rev. 9688b163e0da+ 2024-09-02 01:31:15 +0000 (3.3.5-solocraft branch) (Unix, Debug, Static)
```

**Comment by acidmanifesto** [comment]
> I can't reproduce this issue anymore.
> 
> I can enter instance with both players and bots (npcbots) without disconnection/crash. I'll try to perform some more tests.
> 
> Patch used : https://patch-diff.githubusercontent.com/raw/TrinityCore/TrinityCoreCustomChanges/pull/141.patch
> 
> ```
> trinity@wotlkVM-dev:~/TC_3.3.5-solocraft$ patch  -p1 < 141.patch
> patching file src/server/scripts/Custom/Solocraft.cpp
> patching file src/server/scripts/Custom/Solocraft.cpp
> patching file src/server/scripts/Custom/Solocraft.cpp
> ```
> 
> ```
> TC> server debug
> TC> TrinityCore rev. 9688b163e0da+ 2024-09-02 01:31:15 +0000 (3.3.5-solocraft branch) (Unix, Debug, Static)
> ```
> 
> 

Given the revision you recompiled with my last commit adjustment for a sql query i over looked. It should be good to go unless u have any other issue then please let me know.

**Comment by kruchor** [comment]
I've performed some more tests. Everything is OK now !
Is someone is willing to double check on his side then I think we can close this issue and merge your change.

Thank you all for your quick support and commitment !

**Pull request merged**